src/components/Abstractor/Abstractor.tsx

```diff
@@ -49,26 +67,38 @@ const Abstractor = ({ patientData, library, valueSetMap, questionnaire }: Props)
   const generateQR = () => {
     // Generate QuestionnaireResponse
     const qr = window.LForms.Util.getFormFHIRData('QuestionnaireResponse', 'R4');
-    console.log(qr);
-    // Signify to user that QuestionnaireResponse has been generated
-    let answerCount = 0;
-    if (qr.item) {
-      answerCount = qr.item.length;
-    }
-    setResponseGenerated({ isGenerated:true, count: answerCount});
+    var myJSONString = JSON.stringify(qr, null, 2);
+    return myJSONString;
   }
   
   const generateHL7Message = () => {
     const exporter = window.LForms.Util.getFormHL7Data();
-    console.log(exporter);
+    var myJSONString = JSON.stringify(exporter, null, 2);
+    var myEscapedJSONString = myJSONString.replace(/\\n/g, "\\n")
+                                      .replace(/\\'/g, "\\'")
+                                      .replace(/\\"/g, '\\"')
+                                      .replace(/\\&/g, "\\&")
+                                      .replace(/\\r/g, "\\r")
+                                      .replace(/\\t/g, "\\t")
+                                      .replace(/\\b/g, "\\b")
+                                      .replace(/\\f/g, "\\f");
+
+    return myEscapedJSONString;
```

Comment:
these replaces shouldn't be necessary... the stringify with null and 2 should space things out properly, and using `<pre>` and `<code>` like I mentioned should take care of the display.

```suggestion
    return JSON.stringify(exporter, null, 2);
```